examples/cookbook/app/network-requests/__tests__/PhoneBook.test.tsx

```diff
@@ -0,0 +1,118 @@
+import { render, screen, waitForElementToBeRemoved } from '@testing-library/react-native';
+import React from 'react';
+import axios from 'axios';
+import PhoneBook from '../PhoneBook';
+import { User } from '../types';
+
+jest.setTimeout(10000);
+
+describe('PhoneBook', () => {
+  it('fetches contacts successfully and renders in list', async () => {
+    (global.fetch as jest.Mock).mockResolvedValueOnce({
```

Comment:
One problem with this implementation is that we don't test that the correct url is called here. Also if there were multiple api calls on the same screen and you wanted to test that screen, how would you test that? I feel like eventually you'd end up with a custom api mock library instead of using one

Reply:
Thanks for the review @pierrezimmermannbam! I understand the importance of mocking correctly, especially ensuring the URL matches the expected one. I've gone ahead and added that check in the latest commit.

Regarding the mocking approach, my intention was to present it in a basic and bare-bones way to keep things simple for those new to mocking. However, I appreciate the feedback, and I’ve expanded the documentation to include a "Further Reading and Alternatives" section. This section now covers various ways to mock differently, which should provide more options and context for developers with varying experience levels.

Let me know if there are any other changes you'd like to see!

Reply:
Regarding the mocking approach, my intention was to present it in a basic and bare-bones way to keep things simple for those new to mocking. However, I appreciate the feedback, and I’ve expanded the documentation to include a "Further Reading and Alternatives" section. This section now covers various ways to mock differently, which should provide more options and context for developers with varying experience levels.

Let me know if there are any other changes you'd like to see!

Reply:
I understand your approach but I personally would not recommend not using a library for this because libraries help enforcing good practices and are also easier to use than a custom setup imo and I think we should only document what we consider to be good practices, wdyt @mdjastrzebski ?

Reply:
Thanks for your input, @pierrezimmermannbam. I completely agree that following best practices is crucial, and libraries can definitely help enforce those. That said, I also believe that there are multiple ways to achieve a goal, and implementing a custom solution is a valid approach, especially when starting with something basic and straightforward. In my examples, I aimed to leverage existing APIs to keep things simple and accessible for beginners. Using a library can certainly be a valuable expansion, and it's something that developers can consider as they grow more comfortable with mocking. I've also have mentioned this in the "Further Reading and Alternatives" section.